### ISSUE TITLE
[MU4 Issue] Screen readers may read "not selected" when navigating instrument list (NVDA only)

### ISSUE THREAD

**Issue by MarcSabatella** [musescore/MuseScore#15521]
**Describe the bug**
Blind users navigating the instrument list (available instruments within a family) hear "not selected", which is confusing to them

**To Reproduce**
1. Enable screen reader (from @DmitryArefiev : only NVDA reads "not selected" on my side, other screen readers are fine)
2. Create a new score
3. Select "Choose instruments"
4. Select Woodwinds
5. Press Tab to move to the the Instruments column
6. Press Down repeatedly
7. Hear error: depending on your screen reader, each instrument may report that it is "not selected"

**Expected behavior**
The name of the instrument should be read without that additional text

 - OS: Windows

**Additional context**
I believe this happens with NVDA but not most other screen readers.  It ends up causing a lot of confusion, because hearing it read "not selected" even after the instrument is actually added to the score makes people think it wasn't in fact added even though it was.  I've heard this feedback from three different users.

**Comment by MarcSabatella** [comment]
@Tantacrul @shoogle I know this had been reported multiple times before, but I'm not finding a relevant GitHub issue, so I'm opening this new one.

**Comment by Tantacrul** [comment]
I guess the screen reader should be saying something like 'Add violin (solo)', 'add violins (section)', etc. as you scroll through potential instruments.

@shoogle - I'll pop this on both of us for a design.

**Comment by MarcSabatella** [comment]
In designing that, you might also want to look at the screen reader feedback while navigating the third column.  Right now it reads the names of the instruments in your score, but doesn't (Orca, anyhow) read anything to let you know if you've *selected* the instrument you are currently on.  You need to press Enter to select before you can delete with the Delete kley, but this isn't obvious, nor is it obvious if the instrument is in fact selected, nor is it obvious if the Delete actually worked.  The move buttons also require you to actual select the instrument to enable.  I'm not sure if there really should be a need to explicitly select or not, or if it would make sense to enable deleting an moving right away.

Anyhow, while "Add" is indeed the only action possible in the second column - so adding that word to the screen reader feedback seems logical enough, that same approach wouldn't work in the third column, since there are multiple actions possible.  So a consistent approach across both columns would be nice.

**Comment by Eism** [comment]
@MarcSabatella 
Please check the build from this PR https://github.com/musescore/MuseScore/pull/16504

Actually not quite what you expect, but better than it is at the moment.

My PR fixes the following problems:
1. There is no distinction between selected and unselected states - the screen reader understands which item is selected. Actually, the behavior is different for Orca and NVDA. One says only not selected state, other says only selected state
2. The user does not understand which list he is in - the screen reader is now reading the name of the list

**Comment by DmitryArefiev** [comment]
> @MarcSabatella Please check the build from this PR #16504
> 
> Actually not quite what you expect, but better than it is at the moment.
> 
> My PR fixes the following problems:
> 
>     1. There is no distinction between selected and unselected states - the screen reader understands which item is selected. Actually, the behavior is different for Orca and NVDA. One says only not selected state, other says only selected state
> 
>     2. The user does not understand which list he is in - the screen reader is now reading the name of the list

I've checked #16504 on Narrator,NVDA, VoiceOver and Orca

In 4.2 master build, "not selected" issue happens only with NVDA

About **№1**:
 #16504 fixes only part of "not selected" issue: NVDA reads "not selected" when return to selected element

Comparing **master** and **PR**:


https://github.com/musescore/MuseScore/assets/90187801/908c1e8a-3b41-4a42-a4ec-c2c23fd3c919




About **№1**:
Yes, after #16504, screen reader reads the name of the list (@Eism  - strange, but it **doesn't work in Orca** on my side)

**Comment by MarcSabatella** [comment]
Thanks, definitely seems improved!  I tested with orca, no "not selected" at all, which helps.  It's still likely to be a generally confusing dialog to blind users I'm afraid, but at least less misleading than before.

I do wonder about the change to focus in the search box.  I see this was specifically requested in https://github.com/musescore/MuseScore/issues/10791, and it makes to me as a sighted user.  I wonder if automatically focusing on an element in the "middle" of the dialog is really a great idea for accessibiltiy.  @Tantacrul did we get feedback on this from any blind users?  Now they need to know that there are other things they can only access by navigating backwards.  That's kind of unusual, although I'm sure it's not unheard of.  Maybe people will generally like it because it makes search easier.

Unfortunately, see https://github.com/musescore/MuseScore/issues/19052 - this would need to be fixed (presumably not hard) in order for instrument search to be viable for blind users.

**Comment by DmitryArefiev** [comment]
The same was logged for **Instrument panel** #19927

And also that happens in **Parts dialog**

**Comment by jmwmusic** [comment]
I wonder if some of these dialogues with less standard navigation (usual expectation for most screen readers is to use Tab and land on individual action buttons) where it could be assisted with hints spoken after the item. e.g. in an instrument list after the instrument name could it say "press space to select" and then when you press the space "selected - press delete to remove instrument, use up/down arrows to move" or whatever the action is.  As long as this extra wordage is after the key thing being listened for, in this case the instrument name it won't slow a screen reader down moving ahead quickly on the list.   I think overall the issue most screen users are having in many areas is not knowing things are there as they get jumped over when using Tab or as Marc pointed out we don't know they are selected and realise we have to select before other options, which also may not be obvious are there.  so just a thought that it might be easier in many instances to add hints rather than redesign?

**Comment by jmwmusic** [comment]
I have been testing the Choose Instrument area and had an issue to create regarding no remove button.  However, it seems to fit within this issue.  As a screen reader I can sometimes find the remove button with arrows after adding one to the Orchestration list.  But I can't work out steps of how I get there.  I know I have to have selected the instrument with space or enter, and then very unintuitively a right arrow will get me to make solo and then some combo of various arrows which I can't make sense of or repeat will eventually find me a remove button.  Most users would not even think to look for it like this, which is why it had been reported to me as not being there at all.  These actions really need to be tabable buttons after selecting the instrument.

**Comment by MarcSabatella** [comment]
When you hit Tab after selecting an instrument from the "Your score" list, you are taken to a series of buttons as follows:

- Move up (moves the selected instrument higher in the score)
- Move down (moves the selected instrument lower in the score)

However, the "Remove" button is not part of that group.  It is instead found at the *top* of the list, along with the score order dropdown (which selects between different preset isntrument orderings for different genres of music).  So, for better or for worse, the way you get there after selecting an instrument is to cursor back up to the top of the list, then once more to reach to the score order dropdown, then press "Right" to get to the remove button.

Visually, I don't know that it really makes sense to have Remove at the top, but it's kind of consistent with some other panels like the palettes.  But even if that doesn't change, I do think altering the tab order to group the remove button along with the move up/down button would make sense.

The "Make soloist" button is current found by pressing "Right" immediately after selecting the instrument.  This is indeed how it is laid out visually.  But here too, I don't see a good reason why it shouldn't be grouped with the other controls that operate on the currently selected instrument(s).
